core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -336,14 +337,31 @@ public void validate() {
     if (!getBoolean(PropertyKey.CONF_VALIDATION_ENABLED)) {
       return;
     }
+    List<String> removalMessages = new ArrayList<>();
     for (PropertyKey key : keySet()) {
       Preconditions.checkState(
           getSource(key).getType() != Source.Type.SITE_PROPERTY || !key.isIgnoredSiteProperty(),
           "%s is not accepted in alluxio-site.properties, "
               + "and must be specified as a JVM property. "
               + "If no JVM property is present, Alluxio will use default value '%s'.",
           key.getName(), key.getDefaultValue());
+
+      if (PropertyKey.isDeprecated(key) && getSource(key).compareTo(Source.DEFAULT) > 0) {
```

Comment:
can we change the `> 0` to ` != 0`? Then we don't need to assume anything about the order of the sources.